addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlUtils.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.paramminer;
+
+import java.io.BufferedReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/** A utility class for URL bruteforce related operations for the paramminer add-on. */
+public class UrlUtils {
+    private static Logger logger = LogManager.getLogger(UrlUtils.class);
+    /**
+     * Returns a List of all parameters to be used from a given wordlist file.
+     *
+     * @param path the path to the wordlist.
+     * @return a List<String> of all parameters to be used from a given wordlist file.
+     */
+    public static List<String> read(Path path) {
+        List<String> params = new ArrayList<>();
+        try (BufferedReader br = Files.newBufferedReader(path, StandardCharsets.UTF_8)) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                params.add(StringUtils.strip(line));
+            }
+        } catch (Exception e) {
+            logger.error(e);
+        }
+        return params;
+    }
+
+    /**
+     * Returns a Map of all parameters to be used from a given wordlist along with a fuzzy number
+     * key.
+     *
+     * @param list the List of all parameters to be used from a given wordlist.
+     * @return a Map<String, String> of all parameters to be used from a given wordlist along with a
+     *     fuzzy number key.
+     */
+    public static Map<String, String> populate(List<String> list) {
+        Map<String, String> map = new HashMap<>();
+        String key;
+        String s;
+        for (int i = 0; i < list.size(); i++) {
+            s = list.get(i);
+            String fuzzy = Integer.toString(i);
+            key = StringUtils.repeat("1", (6 - fuzzy.length())) + fuzzy;
+            map.put(s, key);
+        }
+        return map;
+    }
```

Comment:
The use of "key" here still bugs me. Since it's being used as a value.

I'd still suggest 's' be "word" or "paramName" and 'key' be "value".

Reply:
Okay

Reply:
Done